controllers/applicationvolumereplication_controller.go

```diff
@@ -195,658 +276,895 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
-		return unpause && (errorswrapper.Is(err, ErrSameHomeCluster))
+			return requeue
+		}
 	}
 
-	a.log.Info("Found new home cluster", "name", newHomeCluster)
+	requeue = !done
+	a.log.Info("Completed processing placement", "requeue", requeue)
 
-	mwName := BuildManifestWorkName(subscription.Name, subscription.Namespace, MWTypePV)
+	return requeue
+}
 
-	pvMW, err := a.findManifestWork(mwName, newHomeCluster)
-	if err != nil {
-		a.log.Error(err,
-			fmt.Sprintf("Failed to find 'PV restore' ManifestWork for subscription %s", subscription.Name))
-		// DON'T unpause
-		return !unpause
-	}
+func (a *AVRInstance) processPlacement() (bool, error) {
+	a.log.Info("Process AVR Placement", "name", a.avrPlacementRule)
 
-	if pvMW != nil {
-		a.log.Info(fmt.Sprintf("Found manifest work (%v)", pvMW))
-		// Unpause if the ManifestWork has been applied
-		return IsManifestInAppliedState(pvMW)
+	switch a.instance.Spec.Action {
+	case rmn.ActionFailover:
+		return a.runFailover()
+	case rmn.ActionFailback:
+		return a.runFailback()
+	case rmn.ActionRelocate:
+		return a.runRelocate()
 	}
 
-	return a.cleanupAndRestore(subscription, newHomeCluster)
+	// Not a failover, a failback, or a relocation.  Must be an initial deployment.
+	return a.runInitialDeployment()
 }
 
-func (a *AVRInstance) cleanupAndRestore(
-	subscription *subv1.Subscription,
-	newHomeCluster string) bool {
-	const unpause = true
+func (a *AVRInstance) runInitialDeployment() (bool, error) {
+	a.log.Info("Running initial deployment")
+	a.resetDRState()
 
-	pvMWName := fmt.Sprintf(ManifestWorkNameFormat, subscription.Name, subscription.Namespace, MWTypePV)
+	const done = true
 
-	err := a.deleteExistingManifestWork(subscription, pvMWName)
-	if err != nil {
-		a.log.Error(err, "Failed to delete existing PV manifestwork for subscription", "name", subscription.Name)
+	// 1. Check if the user wants to use the preferredCluster
+	homeCluster := ""
+	homeClusterNamespace := ""
 
-		return !unpause
+	if a.instance.Spec.PreferredCluster != "" {
+		homeCluster = a.instance.Spec.PreferredCluster
+		homeClusterNamespace = homeCluster
 	}
 
-	vrgMWName := fmt.Sprintf(ManifestWorkNameFormat, subscription.Name, subscription.Namespace, MWTypeVRG)
-
-	err = a.deleteExistingManifestWork(subscription, vrgMWName)
-	if err != nil {
-		a.log.Error(err, "Failed to delete existing VRG manifestwork for subscription", "name", subscription.Name)
+	if homeCluster == "" && len(a.avrPlacementRule.Status.Decisions) != 0 {
+		homeCluster = a.avrPlacementRule.Status.Decisions[0].ClusterName
+		homeClusterNamespace = a.avrPlacementRule.Status.Decisions[0].ClusterNamespace
+	}
 
-		return !unpause
+	if homeCluster == "" {
+		return !done, fmt.Errorf("PreferredCluster not set and unable to find home cluster in AVRPlacementRule (%s)",
+			a.avrPlacementRule.Name)
 	}
 
-	err = a.restorePVFromBackup(subscription, newHomeCluster)
+	// We have a home cluster
+	err := a.updateUserPlRuleAndCreateVRGMW(homeCluster, homeClusterNamespace)
```

Comment:
(future) Inadvertent AVR spec change (by the user) removing the action field (making it empty) will move the workload back to preferredCluster without any coordination.

As we have 2 spec fields, maybe we should ensure both are empty as a prevention scheme here?

Or this may need more introspection on how to decide we are halfway through a failover or failback and hence decide what and how to accommodate the new spec change.